### PR TITLE
Update Kotlin Native targets

### PR DESCRIPTION
**Pull request by serras**
According to the recently updated [Kotlin Native support](https://kotlinlang.org/docs/native-target-support.html), these are the targets we should support as a library.

### CODE REVIEW

**Review by JavierSegoviaCordoba — APPROVED** [review]
Nice! I didn't know this one

File: arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts [hunk 1]
```diff
@@ -41,62 +41,77 @@ if (isKotlinMultiplatform) {
...
-    tvosX64()
-    watchosArm32()
-    watchosArm64()
+    iosArm64() // soon in tier 1
+    
+    // TIER 2
+    linuxArm64()
     watchosSimulatorArm64()
     watchosX64()
-    watchosX86()
```
Comment (nomisRev) [inline arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts:61]: Should just do a _hard_ remove targets like this 🤔
I can see on sonatype that we get very little downloads for `watchosX86`, but we're getting some... Same for `mingwX64`. These targets are getting < 500 downloads every month, so arguably its all toy projects.

I think there are 3 options:
 - Follow same deprecation - removal process as Kotlin itself
 - Remove in 2.x.x, regardless of what Kotlin is doing
 - Hard remove in 1.2.0

Comment (JavierSegoviaCordoba) [reply arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts:61]: Probably `watchosX86` projects are not using that target; they are just targeting all targets that exist as coroutines library does (and other JetBrains libraries).

And that is the real problem, when we remove those targets all of those libraries will break. I would keep this PR as a draft and merge it in the future following Kotlin (and probably coroutines) practices.

**Review by nomisRev — APPROVED** [review]
Statistics from sona for remove targets for 1.1.0, 1.1.2 and 1.1.3
 - `iosArm32Main`: < 200 in December
 - `watchosX86`: < 200 in December

@serras can you confirm these are the only removed targets? I am approving this PR, so we can hard remove the targets in 1.2.0.

File: arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts (hunk 1 above)
Comment (serras) [reply arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts:61]: What do you think about not removing any target, but adding the new ones? They won’t be removed until 1.9 at least.

Comment (serras) [reply arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts:61]: @JavierSegoviaCordoba I think we should merge (and release a new version of this library) now, so we can start targeting those targets in 1.8.

Comment (JavierSegoviaCordoba) [reply arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts:61]: Up to you :)

Comment (nomisRev) [reply arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts:61]: If we're not removing anything then it's definitely not an issue. We need to make sure to update the CI workflow of Arrow when we update.

Comment (nomisRev) [reply arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts:61]: > Follow same deprecation - removal process as Kotlin itself
> Remove in 2.x.x, regardless of what Kotlin is doing

Then we can follow either of these two strategies, whichever comes first.

**Comment by serras** [comment]
@JavierSegoviaCordoba @nomisRev should I create a new tag to start publication in Maven Central?

**Comment by nomisRev** [comment]
@serras I released 0.12.0-rc1. So we can test it on a PR, I’ll show you tomorrow how we trigger releases. (It works through triggering the generate-tag workflow with the desired parameters).

Thank you this PR 🙏 Great work 🥳

**Pull request merged**